addons/stock/models/stock_location.py

```diff
@@ -197,10 +191,10 @@ def _check_replenish_location(self):
                 if replenish_wh_location:
                     raise ValidationError(_('Another parent/sub replenish location %s exists, if you wish to change it, uncheck it first', replenish_wh_location.name))
 
-    @api.constrains('scrap_location')
+    @api.constrains('usage')
     def _check_scrap_location(self):
         for record in self:
-            if record.scrap_location and self.env['stock.picking.type'].search_count([('code', '=', 'mrp_operation'), ('default_location_dest_id', '=', record.id)], limit=1):
+            if record.usage == 'inventory' and self.env['stock.picking.type'].search_count([('code', '=', 'mrp_operation'), ('default_location_dest_id', '=', record.id)], limit=1):
                 raise ValidationError(_("You cannot set a location as a scrap location when it is assigned as a destination location for a manufacturing type operation."))
```

Comment:
we should probably update this error message to be:

`You cannot set a location as a inventory loss location when it is assigned as a destination location for a manufacturing type operation.`